File: 2022-2023/sourcecode/lab/lab12/ValueInjectBasicsJava/app/src/main/java/pl/edu/uwr/pum/valueinjectbasicsjava/di/modules/GpuModule.java
package pl.edu.uwr.pum.valueinjectbasicsjava.di.modules;

import dagger.Module;
import dagger.Provides;
import pl.edu.uwr.pum.valueinjectbasicsjava.computer.components.GPU;

@Module
public class GpuModule {

    private int gpuCores;

    public GpuModule(int gpuCores) {
        this.gpuCores = gpuCores;
    }

    @Provides
    GPU provideGpu(){
        return new GPU(gpuCores);
    }
}

File: 2022-2023/sourcecode/lab/lab12/ValueInjectBasicsKotlin/app/src/main/java/pl/edu/uwr/pum/valueinjectbasicskotlin/di/modules/GpuModule.kt
package pl.edu.uwr.pum.valueinjectbasicskotlin.di.modules

import dagger.Module
import dagger.Provides
import pl.edu.uwr.pum.valueinjectbasicskotlin.computer.components.GPU

@Module
class GpuModule(private val gpuCores: Int) {

    @Provides
    fun provideGpu(): GPU{
        return GPU(gpuCores)
    }
}